### PR TITLE
Implement SMB Server Settings Data Source

### PR DESCRIPTION
**Pull request by MangirishK**
# Description
SMB Server Settings Data Source Implementation

# ISSUE TYPE
- Feature Pull Request

##### DATASOURCE NAME
SMB Server Settings

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by MangirishK** [comment]
Please find the screenshots here for acceptance testing and coverage:

AT_Tests
AT_Coverage1
AT_Coverage2

**Comment by MangirishK** [comment]
run e2e test

File: powerscale/provider/smb_server_settings_datasource.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+	filter := config.SmbServerSettingsFilter
+	if filter != nil {
+		scopeStr = filter.Scope.ValueString()
+	}
+	if scopeStr == "" {
+		scopeStr = "effective"
+	}
+
+	var state models.SmbServerSettingsDataSourceModel
+	state.ID = types.StringValue(scopeStr)
```
Comment (taohe1012) [inline powerscale/provider/smb_server_settings_datasource.go:287]: for smb server settings, design to use scope as its ID ?

**Comment by MangirishK** [comment]
run e2e test

**Review by taohe1012 — APPROVED** [review]

**Review by P-Cao — APPROVED** [review]

**Pull request merged**